From a3ea1e6513c7e5111818bcaf3b4e47e76f2ec7f9 Mon Sep 17 00:00:00 2001
From: Purbasha Kar <purbasha.kar@intel.com>
Date: Mon, 6 Jul 2015 12:03:03 +0530
Subject: [PATCH] [BT] prevent crash in ldisc if driver is not probed

If ldisc driver is not probed, a crash when opening TTY.
This happens when CHT T4/ST RVP are used with LnP AOB.

Change-Id: I1bddb6e13a37a7182d7413c7e3f1eca30d464d4c
Signed-off-by: Purbasha Kar <purbasha.kar@intel.com>
---
 drivers/misc/intel_ld/lnp_ldisc.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

Index: linux/drivers/misc/intel_ld/lnp_ldisc.c
===================================================================
--- linux.orig/drivers/misc/intel_ld/lnp_ldisc.c	2015-07-08 14:02:05.853306396 -0700
+++ linux/drivers/misc/intel_ld/lnp_ldisc.c	2015-07-08 14:03:09.333305658 -0700
@@ -422,8 +422,10 @@
 static inline void lbf_serial_get(void)
 {
 	pr_debug("%s\n", __func__);
-	WARN_ON(!intel_lbf_lpm.tty_dev);
-	pm_runtime_get_sync(intel_lbf_lpm.tty_dev);
+	if (intel_lbf_lpm.tty_dev)
+		pm_runtime_get_sync(intel_lbf_lpm.tty_dev);
+	else
+		pr_err("%s: no tty device\n", __func__);
 }
 
 static void lbf_host_enable_work(struct work_struct *work)
@@ -446,8 +448,10 @@
 static inline void lbf_serial_put(void)
 {
 	pr_debug("%s\n", __func__);
-	WARN_ON(!intel_lbf_lpm.tty_dev);
-	pm_runtime_put(intel_lbf_lpm.tty_dev);
+	if (intel_lbf_lpm.tty_dev)
+		pm_runtime_put(intel_lbf_lpm.tty_dev);
+	else
+		pr_err("%s: no tty device\n", __func__);
 }
 
 /* lbf_set_host_wake_state
